src/sql/workbench/browser/editor/resourceViewer/resourceViewerInput.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as nls from 'vs/nls';
+import { EditorInput } from 'vs/workbench/common/editor';
+import { Event, Emitter } from 'vs/base/common/event';
+import { URI } from 'vs/base/common/uri';
+import { ResourceViewerState } from 'sql/workbench/common/editor/resourceViewer/resourceViewerState';
+import { TableDataView } from 'sql/base/browser/ui/table/tableDataView';
+
+export interface ColumnDefinition extends Slick.Column<Slick.SlickData> {
+	name: string;
+}
+
+export class ResourceViewerInput extends EditorInput {
+
+	public static ID: string = 'workbench.editorinputs.resourceviewerinputs';
+	public static SCHEMA: string = 'resource-viewer';
+	private _data: TableDataView<Slick.SlickData>;
+	private _columns: string[] = [];
+	private _state: ResourceViewerState;
+
+	private _onColumnsChanged = new Emitter<Slick.Column<Slick.SlickData>[]>();
+	public onColumnsChanged: Event<Slick.Column<Slick.SlickData>[]> = this._onColumnsChanged.event;
+
+	constructor() {
+		super();
+		this._state = new ResourceViewerState();
+		let searchFn = (val: { [x: string]: string }, exp: string): Array<number> => {
+			let ret = new Array<number>();
+			for (let i = 0; i < this._columns.length; i++) {
+				let colVal = val[this._columns[i]];
+				if (colVal && colVal.toLocaleLowerCase().indexOf(exp.toLocaleLowerCase()) > -1) {
+					ret.push(i);
+				}
+			}
+			return ret;
+		};
+
+		this._data = new TableDataView<Slick.SlickData>(undefined, searchFn, undefined, undefined);
+	}
+
+	public getTypeId(): string {
+		return ResourceViewerInput.ID;
+	}
+
+	public getName(): string {
+		return nls.localize('resourceViewerInput.resourceViewer', "Resource Viewer");
+	}
+
+	public getResource(): URI {
```

Comment:
why do you have `getResource()` and `get resource()`?

Reply:
I ended up removing the resource completely - from looking around a bit I'm not sure we need it since the inputs are always going to be tied to an editor from what I can tell. Do you know of a reason why I'd need this for our case here?